go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -618,6 +655,41 @@ func (cm *ClusterManager) releaseHybridOverlayNodeSubnet(nodeName string, subnet
 	return nil
 }
 
+func (cm *ClusterManager) allocateZoneJoinSubnets(node *kapi.Node) ([]*net.IPNet, error) {
+	cm.zoneIdCacheLock.Lock()
```

Comment:
Don't we also need to protect read accesses to `zoneJoinubnetCache`, e.g., in `sync_nodes()`?